addons/mail/static/tests/owl/hooks/use_store_compare_keys_tests.js

```diff
@@ -0,0 +1,218 @@
+    const comparator = proxyComparatorDeep(this.hashFn, this.isEqual, 0);
+    const obj = this.store.state.obj;
+    assert.notOk(comparator(obj, obj), 'initial state of proxy, always different');
+    assert.ok(comparator(obj, obj), 'unchanged proxy');
+    obj.a = 2;
+    assert.notOk(comparator(obj, obj), 'updated proxy after primitive set');
+    assert.ok(comparator(obj, obj), 'unchanged proxy after primitive set');
+    obj.array = [];
+    assert.notOk(comparator(obj, obj), 'updated proxy after add key');
+    obj.array.push(1);
+    assert.notOk(comparator(obj, obj), 'updated proxy after array push');
+    assert.ok(comparator(obj, obj), 'unchanged proxy after array push');
+});
+
+QUnit.test("proxyComparatorDeep on store proxy: temporal comparison, depth 1", async function (assert) {
+    // this is exactly the same as depth 0 because the proxy is managing the depth
+    assert.expect(7);
+    this.start();
+    const comparator = proxyComparatorDeep(this.hashFn, this.isEqual, 1);
+    const obj = this.store.state.obj;
+    assert.notOk(comparator(obj, obj), 'initial state of proxy, always different');
+    assert.ok(comparator(obj, obj), 'unchanged proxy');
+    obj.a = 2;
+    assert.notOk(comparator(obj, obj), 'updated proxy after primitive set');
+    assert.ok(comparator(obj, obj), 'unchanged proxy after primitive set');
+    obj.c = [];
+    assert.notOk(comparator(obj, obj), 'updated proxy after add key');
+    obj.c.push(1);
+    assert.notOk(comparator(obj, obj), 'updated proxy after array push');
+    assert.ok(comparator(obj, obj), 'unchanged proxy after array push');
+});
+
+QUnit.test("proxyComparatorDeep combined: real use case", async function (assert) {
+    assert.expect(12);
+    this.start();
+    const comparator = proxyComparatorDeep(this.hashFn, this.isEqual, {
+        array: 1,
+    });
+    const obj = this.store.state.obj;
+    assert.notOk(comparator({ a: obj }, { a: obj }), 'initial state of proxy, always different');
+    assert.ok(comparator({ a: obj }, { a: obj }), 'unchanged proxy');
+    obj.a = 2;
+    assert.notOk(comparator(obj, obj), 'updated proxy after primitive set');
+    assert.ok(comparator(obj, obj), 'unchanged proxy after primitive set');
+
+    obj.subObj1 = {};
+    obj.subObj2 = {};
+    assert.notOk(comparator({ array: [obj] }, { array: [obj] }), 'depth 1, initial state of proxy, always different');
+    assert.ok(comparator({ array: [obj] }, { array: [obj] }), 'depth 1, unchanged proxy');
+    assert.notOk(comparator({ array: [obj.subObj1] }, { array: [obj.subObj2] }), 'depth 1, changed array content');
+    assert.notOk(comparator({ array: [obj.subObj1] }, { array: [obj.subObj2] }), 'depth 1, still different objects');
+    assert.notOk(comparator({ array: [obj.subObj1] }, { array: [obj.subObj2, obj.subObj1] }), 'depth 1, changed length');
+    assert.notOk(comparator({ array: [obj.subObj1, obj.subObj2] }, { array: [obj.subObj2, obj.subObj1] }), 'depth 1, wrong order');
```

Comment:
assertions are hard to read, would be easier with multi-line

Reply:
ok